src/AutoRest.CSharp/Common/Generation/Writers/CodeWriterExtensions.cs

```diff
@@ -129,27 +129,60 @@ public static CodeWriter WriteFieldDeclarations(this CodeWriter writer, IEnumera
             return writer.Line();
         }
 
-        public static IDisposable WriteMethodDeclaration(this CodeWriter writer, MethodSignatureBase methodBase, params string[] disabledWarnings)
+        private static CodeWriter WriteMethodModifiers(this CodeWriter writer, MethodSignatureBase methodBase, bool isAbstract)
         {
-            foreach (var disabledWarning in disabledWarnings)
-            {
-                writer.Line($"#pragma warning disable {disabledWarning}");
-            }
-
             writer
                 .AppendRawIf("public ", methodBase.Modifiers.HasFlag(Public))
                 .AppendRawIf("internal ", methodBase.Modifiers.HasFlag(Internal))
                 .AppendRawIf("protected ", methodBase.Modifiers.HasFlag(Protected))
                 .AppendRawIf("private ", methodBase.Modifiers.HasFlag(Private));
 
+            writer.AppendRawIf("abstract ", isAbstract);
```

Comment:
Constructor can't be abstract

Reply:
I moved this into the if statement asserting it is a MethodSignature